rs/registry/canister/src/mutations/subnet.rs

```diff
@@ -227,38 +225,43 @@ impl Registry {
                     let key_id = key_config.key_id.clone();
                     let subnet_id = SubnetId::new(*subnet_id);
                     let nodes = receiver_nodes.iter().copied().collect();
-                    ComputeInitialIDkgDealingsArgs::new(key_id, subnet_id, nodes, registry_version)
+                    ReshareChainKeyArgs::new(key_id, subnet_id, nodes, registry_version)
                 },
             )
             .collect()
     }
 
     /// Helper function to make the request and decode the response for
-    /// `compute_initial_i_dkg_dealings`.
-    async fn get_i_dkg_initializations_from_ic00(
+    /// `reshare_chain_key`.
+    async fn get_chain_key_resharing_from_ic00(
         &self,
-        dealing_request: ComputeInitialIDkgDealingsArgs,
+        dealing_request: ReshareChainKeyArgs,
     ) -> ChainKeyInitialization {
         let response_bytes = call(
             CanisterId::ic_00(),
-            "compute_initial_i_dkg_dealings",
+            "reshare_chain_key",
```

Comment:
I guess this is the "root" change (i.e. everything else more or less follows from this). IIUC, this is more or less a "drop-in" replacement. The only change I see that indicates this might be not entirely drop in is (in the new code) lines 255-261, below.

Reply:
Well, this new function is not only named differently, but it uses different candid types as well.

The old functions used `ComputeInitialIDkgDealingArgs` and `ComputeInitialIDkgDealingsResponse`, whereas the new functions uses `ReshareChainKeyArgs` and `ReshareChainKeyResponse`. 

The new response supports IDKG and DKG type key resharing whereas the old type only supports IDKG key resharing. This shows very clearly in the response, where with the new endpoint, we get an enum and depending on the key type, we need to put the `Dealings` or the `TranscriptRecords` into the `initialization`. With the old code, we would always get `Dealings`.

This is the central change, that enables us to do DKG key resharing.